hushline/settings/registration.py

```diff
@@ -0,0 +1,131 @@
+    registration_codes_required = BooleanField(
+        "Require Registration Codes", validators=[OptionalField()]
+    )
+    submit = SubmitField("Submit", name="registration_codes_required", widget=DisplayNoneButton())
+
+
+class CreateInviteCodeForm(FlaskForm):
+    submit = SubmitField("Create Invite Code", name="create_invite_code", widget=Button())
+
+
+class DeleteInviteCodeForm(FlaskForm):
+    invite_code_id = IntegerField(validators=[DataRequired()])
+    submit = SubmitField("Delete", name="delete_invite_code", widget=Button())
+
+
+def register_registration_routes(bp: Blueprint) -> None:
+    @bp.route("/registration", methods=["GET", "POST"])
+    @admin_authentication_required
+    def registration() -> Tuple[str, int] | Response:
+        toggle_registration_form = ToggleRegistrationForm()
+        toggle_registration_codes_form = ToggleRegistrationCodesForm()
+
+        create_invite_code_form = CreateInviteCodeForm()
+        delete_invite_code_form = DeleteInviteCodeForm()
+
+        status_code = 200
+        if request.method == "POST":
+            # Registration enabled
+            if (
+                toggle_registration_form.submit.name in request.form
+                and toggle_registration_form.validate()
+            ):
+                OrganizationSetting.upsert(
+                    key=OrganizationSetting.REGISTRATION_ENABLED,
+                    value=toggle_registration_form.registration_enabled.data,
+                )
+                db.session.commit()
+                if toggle_registration_form.registration_enabled.data:
+                    flash("👍 Registration enabled.")
+                else:
+                    flash("👍 Registration disabled.")
+                return redirect(url_for(".registration"))
+            # Registration codes required
+            elif (
+                toggle_registration_codes_form.submit.name in request.form
+                and toggle_registration_codes_form.validate()
+            ):
+                OrganizationSetting.upsert(
+                    key=OrganizationSetting.REGISTRATION_CODES_REQUIRED,
+                    value=toggle_registration_codes_form.registration_codes_required.data,
+                )
+                db.session.commit()
+                if toggle_registration_codes_form.registration_codes_required.data:
+                    flash("👍 Registration codes required.")
+                else:
+                    flash("👍 Registration codes not required.")
+                return redirect(url_for(".registration"))
+            # Create invite code
+            elif (
+                create_invite_code_form.submit.name in request.form
+                and create_invite_code_form.validate()
+            ):
+                new_invite_code = InviteCode()
+                db.session.add(new_invite_code)
+                db.session.commit()
+                flash(f"Invite code {new_invite_code.code} created.")
```

Comment:
As a note, our flashes disappear after like 2 seconds (see #718) and this is going to mean users loose their registration codes quite quick after creation.

Reply:
It also displays the registration code in a table on the same page it's loading, so the flash message isn't actually all that important here.